src/kinetics/include/antioch/kinetics_parsing.h

```diff
@@ -51,6 +51,15 @@ namespace Antioch
   template<typename CoeffType, typename VectorCoeffType, typename VectorType>
   void reset_rate( KineticsType<CoeffType,VectorCoeffType> & kin, const VectorType & coefs);
 
+  // ParamType is either VectorCoeffType or CoeffType
+  template <typename CoeffType, typename VectorCoeffType, typename ParamType>
+  void reset_parameter_of_rate(KineticsType<CoeffType,VectorCoeffType> & rate,
+                               KineticsModel::Parameters parameter,
+                               const ParamType & new_value);
```

Comment:
Why passing the `ParamType` as a reference? This seems dangerous. Don't we want to pass by value here?